clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -215,16 +227,113 @@ public NetworkClientDelegate.UnsentRequest toUnsentRequest() {
                     .setMemberId(generation.memberId)
                     .setGroupInstanceId(groupInstanceId)
                     .setTopics(new ArrayList<>(requestTopicDataMap.values())));
-            return new NetworkClientDelegate.UnsentRequest(
+            // TODO: KAFKA-15592
+            NetworkClientDelegate.UnsentRequest resp = new NetworkClientDelegate.UnsentRequest(
                 builder,
-                coordinatorRequestManager.coordinator(),
+                coordinatorRequestManager.coordinator());
+            resp.future().whenComplete(
                 (response, throwable) -> {
-                    if (throwable == null) {
-                        future.complete(null);
-                    } else {
-                        future.completeExceptionally(throwable);
+                    try {
+                        if (throwable == null) {
+                            onResponse(response);
+                        } else {
+                            onError(throwable, resp.handler().completionTimeMs());
+                        }
+                    } catch (Throwable t) {
```

Comment:
I'm not sure if come to that point e.g. do we really need this try/catch.

Reply:
I'd say it's good defensive programming, even if it seems unlikely that onResponse or onError will throw. The exception would otherwise be swallowed.

Reply:
in normal cases, OnResponse would not throw.  During mocking it is possible, because if the user pass in an empty offset for the request and mock the response.  An NPE can be thrown.